// GRPC services for the hosted dashboard

syntax = "proto3";

package ray.rpc;

message NodeInfoEventRequest {
    bytes json_data = 1;
}

message NodeInfoEventReply {
}

message RayletInfoEventRequest {
    bytes json_data = 1;
}

message RayletInfoEventReply {
}

message ErrorInfoEventRequest {
    bytes json_data = 1;
}

message ErrorInfoEventReply {
}

message LogFileEventRequest {
    bytes json_data = 1;
}

message LogFileEventReply {
}

message ProfilingStatusEventRequest {
    bytes json_data = 1;
}

message ProfilingStatusEventReply {
}

message ProfilingInfoEventRequest {
    bytes gcs_data = 1;
}

message ProfilingInfoReply {
}

service DashboardService {
  rpc NodeInfoEvent(NodeInfoEventRequest) returns (NodeInfoEventReply);

  rpc RayletInfoEvent(RayletInfoEventRequest) returns (RayletInfoEventReply);

  rpc ErrorInfoEvent(ErrorInfoEventRequest) returns (ErrorInfoEventReply);

  rpc LogFileEvent(LogFileEventRequest) returns (LogFileEventReply);

  rpc ProfilingStatusEvent(ProfilingStatusEventRequest) returns (ProfilingStatusEventReply);

  rpc ProfilingInfoEvent(ProfilingInfoEventRequest) returns (ProfilingInfoReply);
} 